src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -275,6 +289,37 @@ private long getMeanTimeToRecovery(Run<?, ?> run) {
         return 0;
     }
 
+    private void startTrace(BuildData buildData){
+        if(buildData == null){
+            return;
+        }
+        Tracer tracer = GlobalTracer.get();
+        try (Scope scope = tracer.buildSpan("JenkinsBuild").startActive(false)) {
+            final Span span = scope.span();
+            span.setTag(DDTags.SERVICE_NAME, "jenkins");
+            // Add span to cache in order to retrieve it in the endTrace method.
+            DatadogUtilities.severe(logger, null, "startTrace - Cache key: " + buildData.getBuildId(null));
+            DatadogTraceCache.cache.put(buildData.getBuildId(null), span);
+        }
```

Comment:
```suggestion
        Span span = tracer.buildSpan("JenkinsBuild").withTag(DDTags.SERVICE_NAME, "jenkins").start();
        // Add span to cache in order to retrieve it in the endTrace method.
        DatadogUtilities.severe(logger, null, "startTrace - Cache key: " + buildData.getBuildId(null));
        DatadogTraceCache.cache.put(buildData.getBuildId(null), span);
```
You don't need the scope here since the interesting work is happening after this method returns.
You may consider adding the build or job id as a tag to the span.

Reply:
- Yeah I will add more tags later. 
- I am not sure to truly understand what a scope is then. Will there be a scope created in my case?

Reply:
Scope is something that ties the span to a ThreadLocal for the duration of the scope.  This allows the span to be referenced as the parent by child calls when creating new spans. If you're not making any child calls within the scope, then you don't really need the scope, just the span.